src/Authentication/Services/SystemUserService.cs

```diff
@@ -14,133 +15,94 @@ namespace Altinn.Platform.Authentication.Services
         {
             SystemUser newSystemUser = new()
-            {
-                Id = Guid.NewGuid().ToString(),
+            {                
                 IntegrationTitle = request.IntegrationTitle,
                 ProductName = request.ProductName,
                 OwnedByPartyId = partyId.ToString()
             };
-            theMockList.Add(newSystemUser);
-            return Task.FromResult(newSystemUser);
+
+            Guid insertedId = await _repository.InsertSystemUser(newSystemUser);            
+            SystemUser inserted = await _repository.GetSystemUserById(insertedId);
+            return inserted;
         }
 
         /// <summary>
-        /// Returns the list of SystemUsers this PartyID has registered, including "deleted" ones.
+        /// Returns the list of SystemUsers this PartyID has registered.
         /// </summary>
         /// <returns></returns>
-        public Task<List<SystemUser>> GetListOfSystemUsersPartyHas(int partyId)
+        public async Task<List<SystemUser>> GetListOfSystemUsersPartyHas(int partyId)
         {
             if (partyId < 1)
             {
-                return Task.FromResult<List<SystemUser>>(null);
+                return new List<SystemUser>(null);
             }
 
-            return Task.FromResult(theMockList);
+            return await _repository.GetAllActiveSystemUsersForParty(partyId);
         }
 
         /// <summary>
         /// Return a single SystemUser by PartyId and SystemUserId
         /// </summary>
         /// <returns></returns>
-        public Task<SystemUser> GetSingleSystemUserById(Guid systemUserId)
+        public async Task<SystemUser> GetSingleSystemUserById(Guid systemUserId)
         {
-            SystemUser search = theMockList.Find(s => s.Id == systemUserId.ToString());
+            SystemUser search = await _repository.GetSystemUserById(systemUserId);
             
-            return Task.FromResult(search);
+            return search;
         }
 
         /// <summary>
         /// Set the Delete flag on the identified SystemUser
         /// </summary>
         /// <returns></returns>
-        public Task<int> SetDeleteFlagOnSystemUser(Guid systemUserId)
+        public async Task<int> SetDeleteFlagOnSystemUser(Guid systemUserId)
         {
-            SystemUser toBeDeleted = theMockList.Find(s => s.Id == systemUserId.ToString());
-            toBeDeleted.IsDeleted = true;
-            return Task.FromResult(1);
+            SystemUser toBeDeleted = await _repository.GetSystemUserById(systemUserId);
+            if (toBeDeleted != null) 
+            {
+                await _repository.SetDeleteSystemUserById(systemUserId);
+                return 1;
+            }
+
+            return 0;
         }
 
         /// <summary>
         /// Replaces the values for the existing system user with those from the update 
         /// </summary>
         /// <returns></returns>
```

Comment:
you have probably defined the summary,params, returns in the ISystemUserService?

Reply:
ok